process/graduation_criteria.md

```diff
@@ -0,0 +1,115 @@
+  - TODO: create our own project proposal process based on the CNCF project proposal process
+- Begin working toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Gated PRs with unit tests
+    - MSFT code of conduct
+    - LCA bot enabled
+
+## Incubating Stage
+An Incubating project is one that has met all of the requirements of the Sandbox stage and is showing accelerating 
+adoption.
+
+### Entry Requirements
+- Minimum of 3 customers using the project in a production or pre-production environment
+- Minimum of 2 maintainers
+- Full due diligence of the TOC
+  - A great example is the [CNCF Due Diligence Guidelines](https://github.com/cncf/toc/blob/fc161c13a86ba022277a906a10ff51a568406f7c/process/due-diligence-guidelines.md)
+  - TODO: create our own DD guidelines based on the CNCF DD guidelines
+- Demonstrate a substantial ongoing flow of commits and merged contributions
+- Have a healthy number of issues and pull requests from external contributors
+- Set a clear, documented versioning scheme
+- Show significant progress toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Unit tests and integration tests covering a majority of the codebase
+    - Linting
+    - Static analysis (where appropriate)
+    - An initial security review
+    - Contributing guide
+    - Issue and PR templates
+    - Public project board with prioritization
+    - Milestones for releases
+- Telemetry for errors and general usage
+  - If not possible, provide reasoning for why
+
+## Graduation Stage
+Graduation Stage project is one that has met all of the requirements of the Incubation stage and has reached a level
+of maturity and polish which will delight customers.
+
+### Entry Requirements
+- Minimum of 30 customers using the project in production
+- Minimum of 4 maintainers (at least one from outside AzCU)
+- Semester project road map published and updated
+- Full due diligence of the TOC with an eye to any miss alignment
+- Have achieved and maintained a [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have
+    - Load tests and performance tests
+  - Bonus for achieving Silver or higher
+- Full security review and threat analysis with the Azure Green / Red teams
+- Explicitly define a project governance and committer process. This preferably is laid out in a GOVERNANCE.md file and 
+  references an OWNERS.md file showing the current and emeritus committers.
+- Have a public list of project adopters for at least the primary repo (e.g., ADOPTERS.md or logos on the project 
+  website). For a specification, have a list of adopters for the implementation(s) of the spec.
+- SLA / SLO guidance
+- Support with ICM on-call rotations
```

Comment:
does this really apply to every project?

Reply:
I think we should have an educated CSS and ready support escalation to be a supported MSFT project. For which projects would this not be appropriate?

Reply:
I'd be curious to hear what @craiglpeters thinks about that

Reply:
At the moment, I don't think we should require IcM support to be part of the graduation criteria as we do not currently have paid support for these oss efforts (though this could change in the future). Instead, best effort support is provided to users via GitHub issues by the community.

Reply:
I agree, and I think the viability of that best-effort support model should pair with robust quality/testing criteria.

Reply:
Also, should we comment about the support levels for sandbox and incubation stages?
Even just pointing out no support /best effort github issue support would probably be a good idea.